build/Containerfile.sidecar

```diff
@@ -0,0 +1,26 @@
+# Build the sidecar binary
+FROM golang:1.17 as builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY sidecar/ sidecar/
+COPY api/ api/
+COPY internal/ internal/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux go build -a -o k8s-sidecar sidecar/main.go
```

Comment:
nit: use a full name for the executable, like `csi-addons-sidecar`, no need to include `k8s` in there